core/modules/user/user.module

```diff
@@ -1746,6 +1831,35 @@ function user_admin_paths() {
   return $paths;
 }
 
+/**
+ * Implements hook_user_login_paths().
+ */
+function user_user_login_paths() {
+  return array(
+    'user/login' => TRUE,
+    'user/password' => TRUE,
+    'user/register' => TRUE,
+  );
+}
+
+/**
+ * Implements hook_custom_theme().
+ *
+ * Activates the admin theme for login, register, and reset pages if configured
+ * to use the simple login page. Note this by-passes the "view the
+ * administration theme" permission to show the admin theme to anonymous users.
+ */
+function user_custom_theme() {
+  if (!user_is_login_path() || user_is_logged_in()) {
+    return NULL;
+  }
+  $user_login_appearance = config_get('system.core', 'user_login_appearance');
+  if ($user_login_appearance == 'simplified') {
+    return config_get('system.core', 'admin_theme');
```

Comment:
I think we should allow someone to set a login theme and default to the admin theme if not found.

```suggestion
    $login_theme = config_get('system.core', 'login_theme');
    if (empty($login_theme)) {
      $login_theme = config_get('system.core', 'admin_theme');
    }
    return $login_theme;

Reply:
I've tested this approach and it works well. The UI for the setting could come now or after merge IMO so it doesn't hold things up.